homeassistant/components/harmony/remote.py

```diff
@@ -311,6 +316,16 @@ def new_activity(self, activity_info: tuple) -> None:
         self._state = bool(activity_id != -1)
         self._available = True
         self.async_write_ha_state()
+        async_dispatcher_send(
```

Comment:
We shouldn't dispatch from an entity.

`self._client = HarmonyClient(ip_address=host)` needs to be moved out of the entity if we want to use it for multiple entities.

We can setup subscriber callbacks in the entities that want to consume the callsbacks from it.

An example is in `august/subscriber.py`

Reply:
Really helpful example, thank you! I've moved the HarmonyClient to a centralized data object which allows entities to subscribe for callbacks and centralizes some of the logic for common operations (like changing activities).